### DISCUSSION TITLE
tcp port 3000 redirecting to a agar.io server

### DISCUSSION THREAD

**Discussion by kur00n3k0** [GitSquared/edex-ui#1042] [Questions] [answered]
I've downloaded and executed eDEX-UI and of course i wanted to run some random commands, so i executed an port scanner on my own machine and discovered a open port 3000 telling that is redirecting to a agar.io server, that as far as i know is a game. So after that i made a test by executing `iptables -A INPUT -p tcp --dport 3000 -j DROP` to avoid connections to my computer between this port, and the terminal simple stop working!

The only thing i want to know is why this server is needed to the terminal works and what i write is stored in some place? If it is, how this data is handled and why?

**Comment by GitSquared** [comment, marked as answer]
Your port scanner is trying to interpret the open port but it can be opened for many reasons and agar.io has nothing to do with it in this case.

Edex uses a local websocket server to establish a connection between the user interface and the actual terminal it's wrapping. This server will not accept connections from outside your computer, or other programs.

**Reply by kur00n3k0** [reply]
Nmap is not 100% percent accurate so i fell better to hear that, thanks
